Assignment 5.Rmd

```diff
@@ -1,82 +1,74 @@
-
+#This plot creates a new file - a postscript file that is like a pdf
 ```
 ## Part II
 
 #Regression Tree
 
 We want to see if we can build a decision tree to help teachers decide which students to follow up with, based on students' performance in Assistments. We will create three groups ("teacher should intervene", "teacher should monitor student progress" and "no action") based on students' previous use of the system and how many hints they use. To do this we will be building a decision tree using the "party" package. The party package builds decision trees based on a set of statistical stopping rules.
 
-#Visualize our outcome variable "score"
+#Take a look at our outcome variable "score"
 ```{r}
-
+hist(D1$score)
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
 ```{r}
-D1$advice <- 
+D1$advice <- ifelse(D1$score <=0.4, "intervene", ifelse(D1$score > 0.4 & D1$score <=0.8, "monitor", "no action"))
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+score_ctree <- ctree(factor(advice) ~ prior_prob_count + prior_percent_correct + hints, D1)
 ```
 
 #Plot tree
 ```{r}
-
+plot(score_ctree)
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+There are several ways you could answer this question. The main point woul dbe to justify your answer by referring to the distribution of students in each terminal node. For example, students who ask for more than 12 hints are very likely to need intervention.
+
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
-
-D2 <- 
-
-#Generate predicted advice using the predict() command for new students based on tree generated from old students
-
-D2$prediction <- 
-
+D2 <- read.csv("intelligent_tutor_new.csv", header = TRUE)
+#Generate predicted advice for new students based on tree generated from old students
+D2$prediction <- predict(score_ctree, D2)
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
-
-### To Submit Your Assignment
-
-Please submit your assignment by first "knitting" your RMarkdown document into an html file and then commit, push and pull request both the RMarkdown file and the html file.
-
+```{r}
+#All students in the new sample achieved a score of 1, and therefore would need no action. Therefore the accuracy of your model on the new data would be:
+mean(ifelse(D2$prediction == "no action", 1, 0))
+```
```

Comment:
Overall, great job but try to specify whether the model overfits or whether it is generalizable in terms of the accuracy rate.